explorer/interpreter/interpreter.cpp

```diff
@@ -622,8 +623,10 @@ auto Interpreter::InstantiateType(Nonnull<const Value*> type,
       CARBON_ASSIGN_OR_RETURN(
           Nonnull<const Bindings*> bindings,
           InstantiateBindings(&class_type.bindings(), source_loc));
-      return arena_->New<NominalClassType>(&class_type.declaration(), bindings,
-                                           base);
+      return arena_->New<NominalClassType>(
+          &class_type.declaration(), bindings, base,
+          class_type.vtable() ? std::optional{*class_type.vtable().value()}
```

Comment:
I think we should prefer to use the pointer-like syntax for optionals where practical, because it's more concise, and makes it easier to distinguish the application logic from the mechanics of wrapping and unwrapping. It's also probably more consistent with how the corresponding Carbon code will look, and experimenting with writing Carbon-like code is the whole point of using `optional<Nonnull<T*>>` instead of `T*`.

```suggestion
          class_type.vtable() ? std::optional{**class_type.vtable()}
```